### PR TITLE
Add contributor page (no view yet)

### PR DESCRIPTION
**Pull request by shiraji**
## Issue

#78

## Overview (Required)
Add remote/local logic to get contributors.

For displaying contributors, I will let @srym or others do it. (I will do it if no one does this)
https://github.com/DroidKaigi/conference-app-2017/compare/master...shiraji:78_add_contributor_page?expand=1#diff-4cbabe00489cf45da36d6482d4379adbR77

## Links
N/A

## Screenshot
N/A

### CODE REVIEW

File: app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/ContributorsRepository.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import io.github.droidkaigi.confsched2017.model.Contributor;
+import io.reactivex.Single;
+
+@Singleton
+public class ContributorsRepository {
+
+    private final ContributorsLocalDataSource localDataSource;
+
+    private final ContributorsRemoteDataSource remoteDataSourse;
```
Comment (shiraji) [inline app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/ContributorsRepository.java]: not sure I need to have `ContributorsDataSource`

File: app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/ContributorsRepository.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Inject
+    ContributorsRepository(ContributorsLocalDataSource localDataSource, ContributorsRemoteDataSource remoteDataSource) {
+        this.localDataSource = localDataSource;
+        this.remoteDataSourse = remoteDataSource;
+        this.cachedContributors = new LinkedHashMap<>();
+        this.isDirty = true;
+    }
+
+    public Single<List<Contributor>> getContributors() {
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/ContributorsRepository.java]: I think `findAll()` is better name.

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java [hunk 3]
```diff
@@ -41,7 +63,17 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+        super.onStop();
+        compositeDisposable.dispose();
+    }
+
     private void initView() {
         //
     }
+
+    private void renderContributors(List<Contributor> contributors) {
+        // TODO render contributors
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: On next PR, can you bind `LIst<ContributorViewModel>` (not `List<Contributor>`) to RecyclerView?

File: app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/ContributorsLocalDataSource.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    Single<List<Contributor>> getContributors() {
+        return orma.selectFromContributor()
+                .executeAsObservable()
+                .toList()
+                .subscribeOn(Schedulers.io());
+    }
+
+    void updateAllSync(List<Contributor> contributors) {
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/ContributorsLocalDataSource.java]: You can add `private`.

**Comment by konifar** [comment]
@shiraji Commented. Thank you for contribution! Almost perfect!

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java [hunk 5]
```diff
@@ -31,6 +41,18 @@ public void onAttach(Context context) {
...
     }
 
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Disposable disposable = viewModel.getContributors()
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        this::renderContributors,
+                        throwable -> Log.e(TAG, "Failed to find contributors.", throwable)
```
Comment (jmatsu) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: You can use `Timber` to logging.

**Comment by shiraji** [comment]
Do you want me to rebase the commits? @konifar

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java (hunk 3 above)
Comment (shiraji) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: I'm kinda lost here. Could you explain more? You want me to create `ContributorViewModel` for each grid. And then handle click event there? Maybe need to remove [callback](https://github.com/DroidKaigi/conference-app-2017/blob/master/app/src/main/java/io/github/droidkaigi/confsched2017/viewmodel/ContributorsViewModel.java#L12)?

File: app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/ContributorsRemoteDataSource.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package io.github.droidkaigi.confsched2017.repository.sessions;
```
Comment (shiraji) [inline app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/ContributorsRemoteDataSource.java]: package should be `io.github.droidkaigi.confsched2017.repository.contributors;` not `io.github.droidkaigi.confsched2017.repository.sessions;`

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java (hunk 3 above)
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: Sure.
As you said, Create `ContributorViewModel` which has some fields such as `name`, `imageUrl` and Callback. 
You can see [SessionViewModel](https://github.com/DroidKaigi/conference-app-2017/blob/master/app/src/main/java/io/github/droidkaigi/confsched2017/viewmodel/SessionViewModel.java#L224).

Comment (shiraji) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: Got it.

**Comment by konifar** [comment]
LGTM! 💯

**Comment by konifar** [comment]
Thanks for tackling this task!

**Pull request merged**